pkg/disks/loop_test.go

```diff
@@ -0,0 +1,700 @@
+			defer cancel()
+
+			tmpDir, err := os.MkdirTemp(os.TempDir(), "delete-loop-device-tmp-")
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			defer func() {
+				err := os.RemoveAll(tmpDir)
+				if err != nil {
+					t.Fatal(err)
+				}
+			}()
+
+			devfsDir, err := os.MkdirTemp(os.TempDir(), "delete-loop-device-devfs-")
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			defer func() {
+				err := os.RemoveAll(devfsDir)
+				if err != nil {
+					t.Fatal(err)
+				}
+			}()
+
+			if tc.populateFilesystem != nil {
+				err := tc.populateFilesystem(tmpDir, devfsDir)
+				if err != nil {
+					t.Fatal(err)
+				}
+			}
+
+			expectedCommands := tc.expectedCommands(tmpDir, devfsDir)
+
+			executor := exectest.NewFakeExec(expectedCommands...)
+			deviceSymlinkPath := tc.deviceSymlinkPath(devfsDir)
+			imagePath := tc.imagePath(tmpDir)
+			devicePath := tc.devicePath(devfsDir)
+
+			var expectedErr error
+			if tc.expectedErr != nil {
+				expectedErr = tc.expectedErr(tmpDir, devfsDir)
+			}
+
+			err = DeleteLoopDevice(ctx, executor, deviceSymlinkPath, devicePath, imagePath)
+			if !reflect.DeepEqual(err, expectedErr) {
+				t.Fatalf("expected %v error, got %v", expectedErr, err)
+			}
+			if executor.CommandCalls != len(expectedCommands) {
+				t.Fatalf("expected %d command calls, got %d", len(expectedCommands), executor.CommandCalls)
+			}
+
+			symlinksDir := path.Join(devfsDir, "loops")
+			symlinksDirStat, err := os.Stat(symlinksDir)
+			if err != nil && !errors.Is(err, fs.ErrNotExist) {
+				t.Fatalf("unexpected error on symlinksDir %q stat: %v", symlinksDir, err)
+			}
+
+			symlinksDirExists := symlinksDirStat != nil
+			if !reflect.DeepEqual(tc.expectedDevLoopsDirToExist, symlinksDirExists) {
+				t.Fatalf("expected symlinksDir %q to exists %v, got %v", symlinksDir, tc.expectedDevLoopsDirToExist, symlinksDirExists)
+			}
+
+			if symlinksDirExists {
+				var existingLoopDevicesSymlinks []deviceSymlink
+				err = filepath.WalkDir(symlinksDir, func(fpath string, d fs.DirEntry, err error) error {
+					if err != nil {
+						return err
+					}
+
+					if fpath == symlinksDir {
+						return nil
+					}
+
+					if d.Type()&fs.ModeSymlink == 0 {
+						return fmt.Errorf("expected to find only symlinks in symlinks directory, got %v filemode", d.Type())
+					}
+
+					device, err := os.Readlink(fpath)
+					if err != nil {
+						return fmt.Errorf("can't read symlink %q: %w", fpath, err)
+					}
+
+					existingLoopDevicesSymlinks = append(existingLoopDevicesSymlinks, deviceSymlink{
+						symlinkPath: fpath,
+						devicePath:  device,
+					})
+
+					return nil
+				})
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				expectedDeviceSymlinks := tc.expectedDeviceSymlinks(devfsDir)
+
+				if !reflect.DeepEqual(expectedDeviceSymlinks, existingLoopDevicesSymlinks) {
+					t.Fatalf("expected device symlinks %v, got %v", expectedDeviceSymlinks, existingLoopDevicesSymlinks)
+				}
+			}
+		})
```

Comment:
could it just list the dir and do a deep equal?

Reply:
it's listing directory and doing deep equal? What you had in mind?

Reply:
I was thinking more of listing all the files with `os.ReadDir` and put them into array, then comparing it with an expected array (permission modes and other checks), instead of comparing parts of them while walking the array - but I don't feels strong